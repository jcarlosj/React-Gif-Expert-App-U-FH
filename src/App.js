import React, { useState } from 'react';
import PropTypes from 'prop-types'

/** Component */
import { SearchCategory } from './components/SearchCategory';
import { CategoryList } from './components/CategoryList';

/** Functional Component */
const App = ({ defaultCategory }) => {

    const [ categories, setCategories ] = useState( defaultCategory );

    return (  
        <>
            <h2>GifExpertApp</h2>
            <SearchCategory setCategories={ setCategories } />
            <hr />

            {   categories.map( ( category, idx ) => (
                    <CategoryList 
                        key={ `${ idx }-${ category }`} 
                        category={ category } 
                    />
                )) 
            }
        
        </>
    );
}

App.propTypes = {
    defaultCategory: PropTypes.array
}

App.defaultProps = {
    defaultCategory: [ 'Once Upon a Time' ]
}
 
export default App;