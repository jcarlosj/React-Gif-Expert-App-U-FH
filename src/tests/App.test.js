import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import App from '../App';

describe( '<App />', () => {

    test( 'debe desplegar correctamente', () => {
        
        const wrapper = shallow(
            <App />
        );

        expect( wrapper ).toMatchSnapshot();

    });

    test( 'debe mostrar el listado de categorias', () => {

        const 
            categories = [ 'Mazinger Z', 'Transformers' ],
            wrapper = shallow(
                <App defaultCategory={ categories } />
            );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find( 'CategoryList' ).length ).toBe( categories.length );

    });

});