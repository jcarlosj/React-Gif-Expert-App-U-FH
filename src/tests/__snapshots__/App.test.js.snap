// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> debe desplegar correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <SearchCategory
    setCategories={[Function]}
  />
  <hr />
  <CategoryList
    category="Once Upon a Time"
    key="0-Once Upon a Time"
  />
</Fragment>
`;

exports[`<App /> debe mostrar el listado de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <SearchCategory
    setCategories={[Function]}
  />
  <hr />
  <CategoryList
    category="Mazinger Z"
    key="0-Mazinger Z"
  />
  <CategoryList
    category="Transformers"
    key="1-Transformers"
  />
</Fragment>
`;
